smart-contracts/wasm-chain-integration/src/utils.rs

```diff
@@ -337,6 +337,60 @@ pub fn generate_contract_schema_v2(
     }))
 }
 
+/// Tries to generate schemas for events, parameters, return values, and errors
+/// of methods for a contract with a V3 schema.
+pub fn generate_contract_schema_v3(
+    module_bytes: &[u8],
+) -> ExecResult<schema::VersionedModuleSchema> {
+    let artifact = utils::instantiate::<ArtifactNamedImport, _>(&TestHost, module_bytes)?;
+
+    let mut contract_schemas = BTreeMap::new();
+
+    for name in artifact.export.keys() {
+        if let Some(rest) = name.as_ref().strip_prefix("concordium_event_schema_") {
+            if let Some(contract_name) = rest.strip_prefix("init_") {
```

Comment:
Am I understanding correctly that for a contract `foo`, the exported function will be
```
concordium_event_schema_init_foo
```

I am guessing that the `init` is not necessary. I don't think it is necessary to change this, but I would like to know whether it is necessary for some reason or it was just a choice.

Reply:
The `init` part in the name is technically not necessary but since the event schema is stated as part of the init function definition, it seems nice to include `init` here so it is consistent with how the other schemas are named. I kept `init` in the name for now.

Reply:
Ok. I think technically the reason for this for the parameter and return value is that this is the way to distinguish them from the receive ones.

But since you already have gteh `concordium_event_schema` as a prefix, as opposed to `concordium_function_schema`, it would be unnecerssary.

But this is fine. Good job.